app.go

```diff
@@ -449,22 +456,46 @@ func ValidateApp(opts ...Option) error {
 	return app.Err()
 }
 
+// logBuffer will buffer all messages until a logger has been
+// initialized.
+type logBuffer struct {
```

Comment:
readability nit: this file has gotten pretty large. consider moving logBuffer to a sibling file (log.go?) so that we can also independently unit test it.